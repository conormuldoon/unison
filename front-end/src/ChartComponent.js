import { has } from 'lodash';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { VictoryAxis, VictoryChart, VictoryLine, VictoryTheme, VictoryZoomContainer } from 'victory';
import './App.css';
import { varMapping } from './Util';



const PERCENT = "Percentage";
const CELSIUS = "Celsius";

const PREVAL = "precipitation.value";

/**
 * A component for displaying line graphs.
 * 
 * @component
 * 
 */
function ChartComponent(props) {

  const [yVal, setYVal] = useState(undefined);
  const [yLabel, setYLabel] = useState(undefined);


  useEffect(() => {



    // 'Precipitation','Humidity','Wind Direction','Wind Speed','Cloudiness','Cloud Level','Dew Point','Pressure','Temperature'
    const variRequest = varMapping(props.varCur);

    if (variRequest === "precipitation") {
      if (props.minMax) {
        if (props.index === 0) {
          setYVal("precipitation.minvalue");
        } else if (props.index === 1) {
          setYVal(PREVAL);
        } else {
          setYVal("precipitation.maxvalue");
        }

      } else {
        setYVal(PREVAL);
      }
      setYLabel("Millimetres");

    } else if (variRequest === "humidity") {
      setYVal(variRequest);
      setYLabel(PERCENT);
    } else if (variRequest === "windDirection") {

      // Displaying sin of the angle on the graph, rather than the angle/
      setYVal("windDirection.sinDeg");
      setYLabel("sin of angle");
    } else if (variRequest === "windSpeed") {
      setYVal("windSpeed.mps");
      setYLabel("Metres per second");
    } else if (variRequest === "cloudiness") {
      setYVal(variRequest);
      setYLabel(PERCENT);
    } else if (variRequest === "cloudLevel") {
      if (props.index === 0) {
        setYVal("cloud.low");
      } else if (props.index === 1) {
        setYVal("cloud.medium");
      } else {
        setYVal("cloud.high");
      }
      setYLabel(PERCENT);
    } else if (variRequest === "dewPoint") {
      setYVal(variRequest);
      setYLabel(CELSIUS);
    } else if (variRequest === "pressure") {
      setYVal(variRequest);
      setYLabel("Hectopascals");
    } else if (variRequest === "temperature") {
      setYVal(variRequest);
      setYLabel(CELSIUS);
    }


  }, [props.varCur, props.minMax, props.index]);




  if (yVal && props.data && has(props.data[0], yVal)) {

    return (

      <VictoryChart style={{ parent: { maxWidth: "55%", maxHeight: "55%" } }} responsive={true} padding={{ left: 70, bottom: 50, top: 10 }}
        domainPadding={{ x: [0, 0], y: 0 }}
        theme={VictoryTheme.material}
        scale={{ x: "time" }}
        containerComponent={
          <VictoryZoomContainer
            zoomDimension="x"
            zoomDomain={props.zoomDomain}
            onZoomDomainChange={props.handleZoom}
          />
        }
      >
        <VictoryLine
          style={{
            data: { stroke: "#c43a31" },
            parent: { border: "1px solid #ccc" }
          }}
          data={props.data}
          x="date"
          y={yVal}
        />

        <VictoryAxis fixLabelOverlap={true} />
        <VictoryAxis dependentAxis label={yLabel} style={{ axisLabel: { padding: 45 } }} tickFormat={
          (x) => { if (x < .001) { return x.toFixed(3); } if (x > 999) { return Math.round(x); } return x; }} />

      </VictoryChart>

    );
  } else {

    return false;
  }

}


ChartComponent.propTypes = {
  /** Specifies the beginning and end times of the time domain of the x-axis. */
  zoomDomain: PropTypes.object,

  /** Called when there is a zoom domain change. */
  handleZoom: PropTypes.func.isRequired,

  /** Data to be displayed on the graph. */
  data: PropTypes.array,


  /** The weather variable currently selected. */
  varCur: PropTypes.string.isRequired,

  /** True if the if displaying tabs for the precipitation variable.  */
  minMax: PropTypes.bool,

  /** The tab index - 0 for the left tab, 1 for the middle tab, and 2 for the right tab. */
  index: PropTypes.number,

}

export default ChartComponent;


