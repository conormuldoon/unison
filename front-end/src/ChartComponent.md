ChartComponent example:

```js

const data=[{"date":"2019-02-23T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T20:00:00.000+0000","precipitation":{"value":0.5,"minvalue":0.0,"maxvalue":0.7}},{"date":"2019-02-23T21:00:00.000+0000","precipitation":{"value":0.6,"minvalue":0.0,"maxvalue":0.6}},{"date":"2019-02-23T22:00:00.000+0000","precipitation":{"value":3.0,"minvalue":0.0,"maxvalue":0.4}},{"date":"2019-02-23T23:00:00.000+0000","precipitation":{"value":0.2,"minvalue":0.0,"maxvalue":0.5}},{"date":"2019-02-26T18:00:00.000+0000","precipitation":{"value":0.7,"minvalue":0.0,"maxvalue":0.8}},{"date":"2019-02-25T12:00:00.000+0000","precipitation":{"value":0.3,"minvalue":0.0,"maxvalue":0.5}}];


<ChartComponent data={data} yVal={'precipitation.value'} yLabel={'Millimetres'} />


```  
