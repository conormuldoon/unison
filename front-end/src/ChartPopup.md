ChartPopup example:

```js
import {useState} from 'react';
const [display,setDisplay] = useState(false);
function toggleDisplay(){
    if(display)setDisplay(false);
    else setDisplay(true);
}

if(display){
    window.scrollTo(0,0);
}

<div>
    
    <button onClick={toggleDisplay}>{(display)?"Hide":"Display"}</button>
    {display&&<ChartPopup closePopup={toggleDisplay} varCur="Precipitation" location="UCD" fromDate="4/10/2019" toDate="5/10/2019" />}
</div>
```