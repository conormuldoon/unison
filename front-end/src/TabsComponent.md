TabsComponent example:

```js
import {TODAY} from './Util';

import {useState} from 'react';
const [display,setDisplay] = useState(false);
function toggleDisplay(){
    if(display)setDisplay(false);
    else setDisplay(true);
}

if(display){
    window.scrollTo(0,0);
}

<div style={{"width":"80vw"}}>
    <button onClick={toggleDisplay}>{(display)?"Hide":"Display"}</button>
    
    {display&&<div id="popupdiv" >
        <button onClick={toggleDisplay}>Hide</button>
        <TabsComponent varCur="Precipitation" location="Oslo" fromDate={TODAY} toDate={TODAY}/>
        
    </div>}
</div>
```