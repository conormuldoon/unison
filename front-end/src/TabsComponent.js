import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import React, { useState,useEffect  } from 'react';
import "react-tabs/style/react-tabs.css";
import ChartComponent from './ChartComponent';

import {LPER,MEDIAN,HPER,LOW,MEDIUM,HIGH,CL} from './Constant';

import PropTypes from 'prop-types';





/**
 * A component that displays a number of tabs for ChartComponents for the phenomenom being tracked. 
 * For exmaple, for cloud level, there is a tab for high, medium, and low level clouds. Otherwise,
 * only a single ChartComponent is displayed.
 *  
 * @component
 */
function TabsComponent(props){

  const [tabs,setTabs]=useState(undefined);


  useEffect(()=>{

    function updateTabs(){
      if(props.minMax){
  
        setTabs([LPER,MEDIAN,HPER]);
    
      }else if(props.varCur===CL){
    
        setTabs([LOW,MEDIUM,HIGH]);
    
      }else{
        setTabs(undefined);
      }
    }

    if(props.data&&props.data.length>0){

        updateTabs(props.minMax,props.varCur===CL);     

      }else{
        
        setTabs(undefined);  
        
      }
    },[props.data,props.varCur,props.minMax]
  );
  

      function tabsPanel(index){
        return (<TabPanel >
          <div id="tabdiv" data-testid='chart' >
            <ChartComponent data={props.data} zoomDomain={props.zoomDomain} index={index}
              handleZoom={props.setZoomDomain}
              varCur={props.varCur} minMax={props.minMax}  />
          </div>
        </TabPanel>
        );
      }

      return (<div  id="chartdiv">
          {tabs&&<Tabs id="tabscomponent" defaultIndex={1}>
          <TabList >
            <Tab>{tabs[0]}</Tab>
            <Tab>{tabs[1]}</Tab>
            <Tab>{tabs[2]}</Tab>
          </TabList>

          {tabsPanel(0)}
          {tabsPanel(1)}
          {tabsPanel(2)}

        </Tabs>}

        {!tabs&&<div data-testid='chart' id="singlevar">
          <ChartComponent data={props.data} zoomDomain={props.zoomDomain} handleZoom={props.setZoomDomain}
            varCur={props.varCur} minMax={props.minMax} />
          </div>
        }
      </div>
    );

}


TabsComponent.propTypes ={
 
  /** An array of data to be displayed on one or more charts. */
  data: PropTypes.array,

  /** The current weather variable selected. */
  varCur: PropTypes.string,

  /** Determines wheather to display tabs for the precipitation weather variable. This is needed in that with
   * some locations and models only provide a single preciption value (no minimum or maximum).
   */
  minMax: PropTypes.bool,

  /** Determines the range of data to be displayed. */
  zoomDomain: PropTypes.object,

  /** A function used to change the zoom domain. */
  setZoomDomain: PropTypes.func,

  
}

export default TabsComponent;
