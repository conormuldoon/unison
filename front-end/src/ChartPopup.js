import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { IoMdClose } from "react-icons/io";
import './App.css';
import { PRECIP } from './Constant';
import TabsComponent from './TabsComponent';
import { expandLink, chartText } from './Util';


const DLEN = 16;

/**
 * A popup that displays a TabsComponent. Once mounted, it connects to the back-end to obtain data
 * for the from date, to date, location, and weather variable selected.
 * 
 * @component
 */
function ChartPopup(props) {

  const [data, setData] = useState(undefined);
  const [zoomDomain, setZoomDomain] = useState(undefined);
  const [minMax, setMinMax] = useState(false);

  useEffect(() => {

    let active = true;

    async function obtainData() {

      const uri = expandLink(props.linksProperty, props.location, props.curVar, props.fromDate, props.toDate);

      const response = await fetch(uri, {
        method: 'GET',
        headers: new Headers({
          'Content-Type': 'application/json'
        })
      });
      const dataArray = await response.json();

      if (active) {
        const n = dataArray.length;

        for (let i = 0; i < n; i++) {
          const dateS = dataArray[i].date;
          dataArray[i].date = new Date(dateS.substring(0, DLEN));

        }

        if (n > 0) {

          const fd = dataArray[0].date;
          const td = dataArray[dataArray.length - 1].date;

          if (props.curVar === PRECIP && dataArray[0].precipitation.minvalue !== undefined) {

            setMinMax(true);
          } else {
            setMinMax(false);
          }

          setData(dataArray);
          setZoomDomain({ x: [fd, td] });
        } else {

          setData(undefined);
          setZoomDomain(undefined);
          setMinMax(false);
        }
      }

    }

    obtainData();
    const cancel = () => active = false;
    return cancel;

  }, [props.linksProperty, props.curVar, props.location, props.fromDate, props.toDate]);

  function handleClick(e) {
    e.stopPropagation();
  }

  const vc = chartText(props.curVar);

  return (
    <div id="popupdiv" data-testid='chart-div' onClick={handleClick}>
      <div id="iicon">
        <IoMdClose onClick={props.closePopup} size={20} color="rgb(192, 57, 43)" />

      </div>
      <center>
        {vc} data from {props.location.name.trim()}

        {data && <TabsComponent curVar={props.curVar} data={data} zoomDomain={zoomDomain} minMax={minMax}
          setZoomDomain={setZoomDomain} />}

      </center>
    </div>
  );


}

ChartPopup.propTypes = {
  /** Specifies the weather variable currently selected. */
  curVar: PropTypes.string.isRequired,

  /** Specifies the properties of the location selected. */
  location: PropTypes.object.isRequired,

  /** Specifies the start date for the data that is to be displayed. */
  fromDate: PropTypes.string.isRequired,

  /** Specifies the end date for the data that is to be displayed. */
  toDate: PropTypes.string.isRequired,

  /** Called when the close icon is clicked. */
  closePopup: PropTypes.func.isRequired,

  linksProperty: PropTypes.object.isRequired,
}

export default ChartPopup;

