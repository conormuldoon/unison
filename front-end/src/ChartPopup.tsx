
import React, { useEffect, useState, MouseEvent } from 'react';
import { IoMdClose } from "react-icons/io";
import './App.css';
import { PRECIP } from './Constant';
import TabsComponent from './TabsComponent';
import { createChartFactory } from './ChartComponent';
import { chartText } from './Util';
import { EmailShareButton, EmailIcon, WhatsappShareButton, WhatsappIcon, TelegramShareButton, TelegramIcon } from 'react-share';
import { DomainPropType } from 'victory'

const DLEN = 16;

const SSIZE = 24;


export function createPopupFactory(uri: string, curVar: string, name: string) {

  return function popupFactory(closePopup: () => void): JSX.Element {

    return <ChartPopup uri={uri} curVar={curVar} name={name} closePopup={closePopup} />

  }
}

export type ChartData = Record<string, unknown>[];

interface PopupProps {
  uri: string;
  curVar: string;
  name: string;
  closePopup: () => void;
}
/**
 * A popup that displays a TabsComponent. Once mounted, it connects to the back-end to obtain data
 * for the from date, to date, location, and weather variable selected.
 * 
 * @component
 */
function ChartPopup({ uri, curVar, name, closePopup }: PopupProps): JSX.Element {

  const [data, setData] = useState<ChartData>([]);
  const [zoomDomain, setZoomDomain] = useState<DomainPropType>({ x: [0, 1] });
  const [minMax, setMinMax] = useState(false);


  useEffect(() => {

    let active = true;

    async function obtainData() {

      const response = await fetch(uri, {
        method: 'GET',
        credentials: 'omit',
        headers: new Headers({
          'Accept': 'application/json'
        })
      });
      const dataArray = await response.json();

      if (active) {
        const n = dataArray.length;


        for (let i = 0; i < n; i++) {
          const dateS = dataArray[i].date;
          dataArray[i].date = new Date(dateS.substring(0, DLEN));

        }

        if (n > 0) {

          const fd = dataArray[0].date;
          const td = dataArray[dataArray.length - 1].date;

          if (curVar === PRECIP && dataArray[0].precipitation.minvalue !== undefined) {

            setMinMax(true);
          } else {
            setMinMax(false);
          }

          setData(dataArray);
          setZoomDomain({ x: [fd, td] });

        } else {

          setData([]);
          setZoomDomain({ x: [0, 1] });
          setMinMax(false);
        }
      }

    }

    obtainData();
    const cancel = () => { active = false };
    return cancel;

  }, [uri, curVar]);

  function handleClick(e: MouseEvent) {
    e.stopPropagation();
  }

  const vc = chartText(curVar);

  const subject = vc + ' data from ' + name.trim();

  const chartFactory = createChartFactory(data, zoomDomain, setZoomDomain, curVar, minMax);

  return (
    <div id="popupdiv" data-testid='chart-div' onClick={handleClick} style={{ textAlign: 'center' }}>


      <div id="iicon">
        <IoMdClose onClick={closePopup} size={20} color="rgb(192, 57, 43)" />


        <EmailShareButton
          url={uri}
          subject={subject}
        >
          <EmailIcon size={SSIZE} />
        </EmailShareButton>

        <WhatsappShareButton
          url={uri}
          title={subject}
        >
          <WhatsappIcon size={SSIZE} />
        </WhatsappShareButton>

        <TelegramShareButton
          url={uri}
          title={subject}
        >
          <TelegramIcon size={SSIZE} />
        </TelegramShareButton>

      </div>


      <div style={{ marginTop: 10, marginBottom: 20, textAlign: 'center' }}>
        {subject}
      </div>


      {data && <TabsComponent curVar={curVar} minMax={minMax} chartFactory={chartFactory} />}



    </div>
  );


}


export default ChartPopup;

