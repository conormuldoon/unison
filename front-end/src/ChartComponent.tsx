import { has } from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import {
  VictoryAxis, VictoryChart, VictoryLine, VictoryTheme, VictoryZoomContainer,
  DomainPropType, VictoryZoomContainerProps
} from 'victory';
import './App.css';
import { varMapping } from './Util';

import { ChartData } from './ChartPopup';


const PERCENT = "Percentage";
const CELSIUS = "Celsius";

const PREVAL = "precipitation.value";


export function createChartFactory(data: ChartData, zoomDomain: DomainPropType,
  handleZoom: (domain: DomainPropType, props: VictoryZoomContainerProps) => void,
  curVar: string, minMax: boolean) {

  return function chartFactory(index: number | undefined): JSX.Element {

    return <ChartComponent data={data} zoomDomain={zoomDomain} index={index}
      handleZoom={handleZoom}
      curVar={curVar} minMax={minMax} />
  }
}

interface ChartProps {

  /**
   * The current weather variable.
   */
  curVar: string;

  /**
   * A flag that is true in cases where preciptation data is ternary.
   */
  minMax: boolean;

  /**
   * Specifies the index of the chart in cases where there are multiple chart tabs.
   */
  index: number | undefined;

  /**
   * The dat that is displayed on the chart.
   */
  data: ChartData;

  /**
   * A two element tuple that specifies the chart's zoom domain.
   */
  zoomDomain: DomainPropType;

  /**
   * A callback invoked when the zoom domain of the chart is changed.
   */
  handleZoom: (domain: DomainPropType, props: VictoryZoomContainerProps) => void;
}

/**
 * A component for displaying line graphs.
 * 
 * @component
 * 
 */
function ChartComponent({ curVar, minMax, index, data, zoomDomain, handleZoom }: ChartProps): JSX.Element | null {

  let yVal;
  let yLabel;

  // 'Precipitation','Humidity','Wind Direction','Wind Speed','Cloudiness','Cloud Level','Dew Point','Pressure','Temperature'
  const variRequest = varMapping(curVar);

  if (variRequest === "precipitation") {
    if (minMax) {
      if (index === 0) {
        yVal = "precipitation.minvalue";
      } else if (index === 1) {
        yVal = PREVAL;
      } else {
        yVal = "precipitation.maxvalue";
      }

    } else {
      yVal = PREVAL;
    }
    yLabel = "Millimetres";

  } else if (variRequest === "humidity") {
    yVal = variRequest;
    yLabel = PERCENT;
  } else if (variRequest === "windDirection") {

    // Displaying sin of the angle on the graph, rather than the angle/
    yVal = "windDirection.sinDeg";
    yLabel = "sin of angle";
  } else if (variRequest === "windSpeed") {
    yVal = "windSpeed.mps";
    yLabel = "Metres per second";
  } else if (variRequest === "cloudiness") {
    yVal = variRequest;
    yLabel = PERCENT;
  } else if (variRequest === "globalRadiation") {
    yVal = variRequest;
    yLabel = "Watts per meter squared";
  } else if (variRequest === "cloudLevel") {
    if (index === 0) {
      yVal = "cloud.low";
    } else if (index === 1) {
      yVal = "cloud.medium";
    } else {
      yVal = "cloud.high";
    }
    yLabel = PERCENT;
  } else if (variRequest === "dewPoint") {
    yVal = variRequest;
    yLabel = CELSIUS;
  } else if (variRequest === "pressure") {
    yVal = variRequest;
    yLabel = "Hectopascals";
  } else if (variRequest === "temperature") {
    yVal = variRequest;
    yLabel = CELSIUS;
  }

 
  
  if (yVal && data.length > 1 && has(data[0], yVal)) {

    return (

      <VictoryChart style={{ parent: { maxWidth: "55%", maxHeight: "55%" } }} padding={{ left: 70, bottom: 50, top: 10 }}
        domainPadding={{ x: [0, 1], y: 0 }}
        theme={VictoryTheme.material}
        scale={{ x: "time" }}
        containerComponent={
          <VictoryZoomContainer
            zoomDimension="x"
            zoomDomain={zoomDomain}
            onZoomDomainChange={handleZoom}
          />
        }
      >
        <VictoryLine
          style={{
            data: { stroke: "#c43a31" },
            parent: { border: "1px solid #ccc" }
          }}
          data={data}
          x="date"
          y={yVal}
        />

        <VictoryAxis fixLabelOverlap={true} />
        <VictoryAxis dependentAxis label={yLabel} style={{ axisLabel: { padding: 45 } }} tickFormat={
          (x) => { if (Math.abs(x) < .001) { return parseFloat(x.toFixed(3)); } if (Math.abs(x) > 999) { return Math.round(x); } return x; }} />

      </VictoryChart>

    );
  } else {

    return null;
  }

}


ChartComponent.propTypes = {
  /** Specifies the beginning and end times of the time domain of the x-axis. */
  zoomDomain: PropTypes.object,

  /** Called when there is a zoom domain change. */
  handleZoom: PropTypes.func.isRequired,

  /** Data to be displayed on the graph. */
  data: PropTypes.array.isRequired,


  /** The weather variable currently selected. */
  curVar: PropTypes.string.isRequired,

  /** True if the if displaying tabs for the precipitation variable.  */
  minMax: PropTypes.bool,

  /** The tab index - 0 for the left tab, 1 for the middle tab, and 2 for the right tab. */
  index: PropTypes.number,

}

export default ChartComponent;


