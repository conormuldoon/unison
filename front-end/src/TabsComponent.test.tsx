
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import TabsComponent from './TabsComponent';
import { createChartFactory } from './ChartComponent';
import { ChartData } from './ChartPopup';

const DLEN = 16;


it('renders without crashing', async () => {

    render(<TabsComponent minMax={false} curVar='Cloudiness' chartFactory={(index) => null} />);
});

it('matches snapshot', () => {


    const { container } = render(<TabsComponent minMax={false} curVar='Cloudiness' chartFactory={(index) => null} />);

    expect(container).toMatchSnapshot();

});



const addChart = async (weatherVariable: string, dataArray: ChartData, zoomDomain: { x: [Date, Date] }) => {


    const chartFactory = createChartFactory(dataArray, zoomDomain, () => { }, weatherVariable, false);

    render(<TabsComponent curVar={weatherVariable} minMax={false} chartFactory={chartFactory} />);


    await screen.findByTestId('chart');

}



it('adds tabs for precipitation', async () => {

    await addChart('Precipitation', [{
        "date": "2019-04-01T23:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0.1
        }
    },
    {
        "date": "2019-04-02T00:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T01:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T02:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T03:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T04:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T05:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T06:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T07:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T08:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T09:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    },
    {
        "date": "2019-04-02T10:00:00.000+0000",
        "precipitation": {
            "value": 0,
            "minvalue": 0,
            "maxvalue": 0
        }
    }], { x: [new Date('2019-04-01T23:00:00.000+0000'.substring(0, DLEN)), new Date('2019-04-02T10:00:00.000+0000'.substring(0, DLEN))] });



});
