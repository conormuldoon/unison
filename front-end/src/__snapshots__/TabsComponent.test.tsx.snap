// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    id="chartdiv"
  >
    <div
      data-testid="chart"
      id="singlevar"
    />
  </div>
</div>
`;
